File: output/ChatGPT/avatar/Python/Java/atcoder_ABC133_B.java
import java.util.Scanner;

public class atcoder_ABC133_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int d = scanner.nextInt();
        
        int[][] points = new int[n][d];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < d; j++) {
                points[i][j] = scanner.nextInt();
            }
        }
        
        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (dist(points[i], points[j]) % 1 == 0) {
                    count++;
                }
            }
        }
        
        System.out.println(count);
    }
    
    public static double dist(int[] x, int[] y) {
        double tmp = 0.0;
        for (int i = 0; i < x.length; i++) {
            tmp += Math.pow(x[i] - y[i], 2);
        }
        return Math.sqrt(tmp);
    }
}

File: output/ChatGPT/avatar/Python/C++/atcoder_ABC133_B.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

double dist(vector<int> x, vector<int> y) {
    double tmp = 0.0;
    for (int i = 0; i < x.size(); i++) {
        tmp += pow(x[i] - y[i], 2);
    }
    return sqrt(tmp);
}

int main() {
    int n, d;
    cin >> n >> d;
    vector<vector<int>> points(n, vector<int>(d));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < d; j++) {
            cin >> points[i][j];
        }
    }

    int count = 0;
    for (int i = 0; i < points.size(); i++) {
        for (int j = i + 1; j < n; j++) {
            if (dist(points[i], points[j]) == floor(dist(points[i], points[j]))) {
                count++;
            }
        }
    }

    cout << count << endl;

    return 0;
}